fdbclient/include/fdbclient/MetaclusterManagement.actor.h

```diff
@@ -1749,6 +1749,23 @@ struct ConfigureTenantImpl {
 			throw invalid_tenant_state();
 		}
 
+		{
+			// For now, disallow updating assigned cluster of a tenant.
+			auto iter = self->configurationParameters.find("assigned_cluster"_sr);
+			if (iter != self->configurationParameters.end() && iter->second.present()) {
+				const auto& newClusterName = iter->second.get();
+				if (!tenantEntry.get().assignedCluster.present() ||
+				    tenantEntry.get().assignedCluster.get() != newClusterName) {
+					TraceEvent("UpdateManagementCluster")
+					    .detail("OriginalAssignedCluster",
+					            tenantEntry.get().assignedCluster.present() ? tenantEntry.get().assignedCluster.get()
+					                                                        : "null"_sr)
+					    .detail("NewAssignedCluster", newClusterName);
+					throw invalid_tenant_configuration();
+				}
+			}
+		}
```

Comment:
We have a loop below where we are processing the parameters (see line 1775), I think it would be reasonable to check for this condition there and avoid having to search them twice.

Reply:
The motivation here is to return early to avoid unnecessary work, but I guess I can move the check there.

Reply:
Done